### ISSUE TITLE
Cannot combine Git files and Git directories generators with matrix

### ISSUE THREAD

**Issue by romuduck** [argoproj/applicationset#475]
Hello,

### Current behavior
When using matrix generator, we cannot combine:
- a Git directory generator which would browse through services for example
- a Git file generator which would look for some cross-service configuration files

Probably obvious, but as both generators output a ```path``` key, but are browsing different sources, it results in following error:
```
msg="error generating params" error="found duplicate key path.basename with different value, a: xxx ,b: yyy" ...
```

### Expectation
I guess we should either update the documentation and/or provide a way to give precedence.

Thanks all

**Comment by crenshaw-dev** [comment]
@romuduck wouldn't a file generator with something like `../**/config.yaml` be the same thing as a file x directory matrix generator?

**Comment by w4rgrum** [comment]
Allowing to prefix the variables in output of the generators would be nice and help using twice the same generator in a matrix generator. This could be done by adding an optional "name" field to generator for example, if present all outputs would be prefixed by "<name>."

**Comment by crenshaw-dev** [comment]
@w4rgrum true, that would work. Do you have a use case that requires a git file * git directory matrix generator?

**Comment by adamjohnson01** [comment]
@crenshaw-dev, I am happy to have a go at implementing this.

**Comment by elebioda** [comment]
I have a use case where our git file generator will grab all our cluster configs that contain certain information: gcpProject, gcpRegion and then use a directory generator to bring up our apps across all clusters.

**Comment by yoav-klein** [comment]
+1, I could use this feature also, any new about that?

**Comment by ppratheesh** [comment]
I would like to use this scenario as well

**Comment by gaeljw** [comment]
> I have a use case where our git file generator will grab all our cluster configs that contain certain information: gcpProject, gcpRegion and then use a directory generator to bring up our apps across all clusters.

I have a very similar use case. Being able to name/prefix generators would be helpful for this.

**Comment by crenshaw-dev** [comment]
Being implemented here: https://github.com/argoproj/argo-cd/pull/10523
